### DISCUSSION TITLE
Production-ready or not?

### DISCUSSION THREAD

**Discussion by strickvl** [pydantic/pydantic#6199] [Question] [answered]
[The Pydantic GH Repo says](https://github.com/pydantic/pydantic#pydantic-v110-vs-v2) that v2 is "not yet ready for production use" whereas [the v2 migration guide on the docs says](https://docs.pydantic.dev/dev-v2/migration/) that "we believe Pydantic V2 is ready for your production applications."

Which of those two is the reality? Thank you!

**Comment by adriangb** [comment, marked as answer]
It's currently a beta. Once a non beta is out that indicates that we think it's ready. But keep in mind that whatever we tell you with version numbers, and frankly whatever anyone else tells you, is just what the software author believes to be true. Semantic versioning is a communication tool, not a guarantee. So even when we release a final version of Pydantic V2 it's not unreasonable for you to say "this is relatively new software, it's a bit too risky for my application, I'm going to wait a bit till it becomes more stable". By the same token, if your application is not mission critical maybe the beta release is production ready for you.

**Reply by samuelcolvin** [reply]
Also worth noting that there are many open source packages which are far lower quality than pydantic V2, and are used in production.

In my experience, "production ready" mostly means "I ran it in production and no one shouted at me".

**Reply by samuelcolvin** [reply]
@adriangb we should probably change the message in the readme, it's pretty out of date.

**Reply by Lithimlin** [reply]
Should people new to pydantic still bother learning v1 or should they start with v2? Is the documentation in its current state sufficient to jump into v2 as a newbie?

**Reply by samuelcolvin** [reply]
The documentation is getting pretty close, and we're hoping to release pydantic V2 before the end of the month, so I would start with pydantic V2.

While getting started, most of the concepts should be the same between v1 and v2.
